### DISCUSSION TITLE
Improve CD support

### DISCUSSION THREAD

**Discussion by MeTalOneOEight** [ZuluSCSI/ZuluSCSI-firmware#161] [Ideas]
Since SCSI2SD only has a simple CD drive implementation and image support, maybe including cdemu 
https://cdemu.sourceforge.io/
https://github.com/cdemu/cdemu
will help with that.
It supports many image formats, uses SCSI commands and is GPL licensed.

**Comment by PetteriAimonen** [comment]
Thanks, this is an useful reference. CDEmu has strong dependencies on GLib and libmirage, which mean that it is not directly possible to use it on the embedded system.

**Reply by MeTalOneOEight** [reply]
Glib seems a real dealbreaker for embedded.
Cross compiling the lib is also not an option I guess.

Maybe using libcdio instead is an option?
More to code yourself though.

**Reply by MeTalOneOEight** [reply]
What are the RAM and Firmware ROM size limits on the different Zulu Boards?
I was looking at why GLib is problematic and read it can be compiled for embedded systems but it's 2MB in size, which is too large for most embedded systems.
If it doesn't fit, rewriting the needed parts of cdemu without Glib is too much of an effort I guess.

An alternative would be GNU libcdio, which also has image support for ISO, cue/bin and toc/bin with way less dependencies.

**Reply by PetteriAimonen** [reply]
The current firmware size is around 150kB and most of the platforms have around 256kB of total space.

I think a reasonable approach would be to figure out which format has the best support for CD-ROM features.
Other formats could then be converted to that on PC.
